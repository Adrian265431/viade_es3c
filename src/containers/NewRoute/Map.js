import { Map, GoogleApiWrapper, Marker, Polyline, HeatMap, InfoWindow } from "google-maps-react";
import * as Papa from "papaparse";
import React from "react";
import update from "react-addons-update";
import axios from "axios";
import { Loader } from "@util-components";

const mapStyle = {
	paddingBottom: "10px",
	height: "100%"
};

const infoWindowStyle = {
	display: "block",
	fontSize: "10px",
	margin: 0
};
var heatMap = [];

var count = 0;
var data = [];
var dataActualizada = [];
var gradient = [
	"rgba(0, 255, 255, 0)",
	"rgba(0, 255, 255, 1)",
	"rgba(0, 191, 255, 1)",
	"rgba(0, 127, 255, 1)",
	"rgba(0, 63, 255, 1)",
	"rgba(0, 0, 255, 1)",
	"rgba(0, 0, 223, 1)",
	"rgba(0, 0, 191, 1)",
	"rgba(0, 0, 159, 1)",
	"rgba(0, 0, 127, 1)",
	"rgba(63, 0, 91, 1)",
	"rgba(127, 0, 63, 1)",
	"rgba(191, 0, 31, 1)",
	"rgba(255, 0, 0, 1)"
];
export class MapContainer extends React.Component {
	sendData = () => {
		this.props.parentCallBack(this.state.markers);
	};
	state = {
		markers: [],
		mapCovid: [],
		currentZoom: null,
		isHeatVisible: false,
		activeMarker: null,
		showInfoWindow: false
	};

	getLocation() {
		if (navigator.geolocation) {
			navigator.geolocation.getCurrentPosition((position) => {
				this.setState({
					center: {
						lat: position.coords.latitude,
						lng: position.coords.longitude
					}
				});
			});
		}
	}

	componentDidMount() {
		this._asyncRequest = axios
			.get("https://raw.githubusercontent.com/microsoft/Bing-COVID-19-Data/master/data/Bing-COVID19-Data.csv")
			.then((response) => {
				data = null;
				this._asyncRequest = null;
				var config = {
					header: true
				};
				data = Papa.parse(response.data, config);
				data = data.data;

				var region = this.groupByArray(data, "AdminRegion2");

				region.map((country = {}) => {
					dataActualizada.push(country.values[country.values.length - 1]);
				});

				var region2 = this.groupByArray(data, "AdminRegion1");
				region2.map((country = {}) => {
					dataActualizada.push(country.values[country.values.length - 1]);
				});

				var pais = this.groupByArray(data, "Country_Region");
				pais.map((country = {}) => {
					dataActualizada.push(country.values[country.values.length - 1]);
				});

				this.iniciateMarkers();
			});
	}

	componentWillUnmount() {
		dataActualizada = [];
		data = [];
		heatMap = [];
		count = 0;
	}

	groupByArray(xs, key) {
		return xs.reduce(function(rv, x) {
			let v = key instanceof Function ? key(x) : x[key];
			let el = rv.find((r) => r && r.key === v);
			if (el) {
				el.values.push(x);
			} else {
				rv.push({ key: v, values: [ x ] });
			}
			return rv;
		}, []);
	}

	iniciateMarkers() {
		if (!this.state.mapCovid.length > 0 && dataActualizada.length > 0 && count === 0) {
			let mapCovid = this.state.mapCovid;
			// eslint-disable-next-line
			const geoJson = {
				type: "FeatureCollection",
				// eslint-disable-next-line
				features: dataActualizada.map((country = {}) => {
					if (country.Latitude && country.Longitude) {
						mapCovid = update(mapCovid, {
							$push: [
								{
									lat: country.Latitude,
									lng: country.Longitude,
									recuperados: country.Recovered,
									muertes: country.Deaths,
									casos: country.Confirmed,
									country_p: country.Country_Region,
									region: country.AdminRegion1,
									city: country.AdminRegion2,
									updated: country.Updated
								}
							]
						});
					}
				})
			};
			this.setState({ mapCovid });
			count = 1;

			this.sendData();
		}
	}

	clickPoint = (event, map, clickEvent) => {
		let { markers } = this.state;
		markers = update(markers, {
			$push: [
				{
					position: {
						lat: clickEvent.latLng.lat(),
						lng: clickEvent.latLng.lng()
					},
					index: this.state.markers.length,
					defaultAnimation: 2,
					key: this.state.markers.length
				}
			]
		});
		this.setState({ markers });
		this.sendData();
	};

	draw() {
		let markers = [];
		for (let i = 0; i < this.state.markers.length; i++) {
			markers.push({
				lat: this.state.markers[parseInt(i)].position.lat,
				lng: this.state.markers[parseInt(i)].position.lng
			});
		}
		return markers;
	}

	onMarkerClick = (props, marker, e) => {
		let markers2 = props.markersList;
		for (var i = 0; i < markers2.length; i++) {
			if (markers2[i].key === props.index) {
				markers2.splice(i, 1);
				this.setState({ markers2 });
				this.sendData();
			}
		}
	};

	handleToggle = () => {
		this.setState({ isHeatVisible: !this.state.isHeatVisible });
	};
	handleMouseOver = (props, marker, e) => {
		this.setState({
			activeMarker: marker,
			showInfoWindow: true
		});
	};

	handleZoom = (props, map, e) => {
		this.setState({
			currentZoom: map.zoom
		});
		console.log(this.state.currentZoom);
	};

	close = () => {
		this.setState({
			activeMarker: null,
			showInfoWindow: false
		});
	};

	render() {
		this.getLocation();

		heatMap = (
			<HeatMap
				visible={this.state.isHeatVisible}
				gradient={gradient}
				opacity={1}
				positions={this.state.mapCovid}
				radius={20}
				center={this.state.center}
				heatmapMode={"POINTS_WEIGHT"}
			/>
		);
		return (
			<Map
				google={this.props.google}
				zoom={5}
				minZoom={3}
				onZoomChanged={this.handleZoom}
				style={mapStyle}
				heatmapLibrary={true}
				onClick={this.clickPoint}
				center={this.state.center}
				gestureHandling={"cooperative"}
				styles={[
					{ elementType: "geometry", stylers: [ { color: "#242f3e" } ] },
					{ elementType: "labels.text.stroke", stylers: [ { color: "#242f3e" } ] },
					{ elementType: "labels.text.fill", stylers: [ { color: "#746855" } ] },
					{
						featureType: "administrative.locality",
						elementType: "labels.text.fill",
						stylers: [ { color: "#d59563" } ]
					},
					{
						featureType: "poi",
						elementType: "labels.text.fill",
						stylers: [ { color: "#d59563" } ]
					},
					{
						featureType: "poi.park",
						elementType: "geometry",
						stylers: [ { color: "#263c3f" } ]
					},
					{
						featureType: "poi.park",
						elementType: "labels.text.fill",
						stylers: [ { color: "#6b9a76" } ]
					},
					{
						featureType: "road",
						elementType: "geometry",
						stylers: [ { color: "#38414e" } ]
					},
					{
						featureType: "road",
						elementType: "geometry.stroke",
						stylers: [ { color: "#212a37" } ]
					},
					{
						featureType: "road",
						elementType: "labels.text.fill",
						stylers: [ { color: "#9ca5b3" } ]
					},
					{
						featureType: "road.highway",
						elementType: "geometry",
						stylers: [ { color: "#746855" } ]
					},
					{
						featureType: "road.highway",
						elementType: "geometry.stroke",
						stylers: [ { color: "#1f2835" } ]
					},
					{
						featureType: "road.highway",
						elementType: "labels.text.fill",
						stylers: [ { color: "#f3d19c" } ]
					},
					{
						featureType: "transit",
						elementType: "geometry",
						stylers: [ { color: "#2f3948" } ]
					},
					{
						featureType: "transit.station",
						elementType: "labels.text.fill",
						stylers: [ { color: "#d59563" } ]
					},
					{
						featureType: "water",
						elementType: "geometry",
						stylers: [ { color: "#17263c" } ]
					},
					{
						featureType: "water",
						elementType: "labels.text.fill",
						stylers: [ { color: "#515c6d" } ]
					},
					{
						featureType: "water",
						elementType: "labels.text.stroke",
						stylers: [ { color: "#17263c" } ]
					}
				]}
			>
				<div id="floating-panel">
					<button id={"buttonCovid"} onClick={this.handleToggle}>
						{" "}
						Covid heatMap
					</button>
				</div>
				{this.state.isHeatVisible ? heatMap : null}

				{this.state.mapCovid.map((point) => {
					return (
						<Marker
							position={{ lat: point.lat, lng: point.lng }}
							c={point}
							cursor={"hand"}
							icon={"http://maps.google.com/mapfiles/ms/icons/red.png"}
							visible={this.state.currentZoom >= 8 && this.isHeatVisible}
							onMouseover={this.handleMouseOver}
							key={point.key}
							tracksViewChanges={false}
						/>
					);
				})}
				{this.state.showInfoWindow ? (
					<InfoWindow
						marker={this.state.activeMarker}
						visible={this.state.showInfoWindow}
						maxWidth={120}
						onClose={this.close}
					>
						<h5 style={infoWindowStyle}>{this.state.activeMarker.c.country_p}</h5>
						<h5 style={infoWindowStyle}>{this.state.activeMarker.c.region}</h5>
						<h5 style={infoWindowStyle}>{this.state.activeMarker.c.city}</h5>
						<p style={infoWindowStyle}> Cases: {this.state.activeMarker.c.casos}</p>
						<p style={infoWindowStyle}> Deaths: {this.state.activeMarker.c.muertes}</p>
						<p style={infoWindowStyle}> Recovered: {this.state.activeMarker.c.recuperados}</p>
					</InfoWindow>
				) : null}

				{this.state.markers.map((marker) => {
					return (
						<Marker
							key={marker.position.lat + marker.position.lng}
							position={{ lat: marker.position.lat, lng: marker.position.lng }}
							icon={"http://maps.google.com/mapfiles/ms/icons/blue.png"}
							markersList={this.state.markers}
							index={marker.key}
							onClick={this.onMarkerClick}
						/>
					);
				})}
				<Polyline path={this.draw()} strokeColor="#01C9EA" strokeOpacity={0.8} strokeWeight={2} />
			</Map>
		);
	}
}

export default GoogleApiWrapper({
	apiKey: "AIzaSyB3kcMOsEbg2unbt5yGvqw4HxNlSLE-U00",
	libraries: [ "visualization" ]
})(MapContainer);
