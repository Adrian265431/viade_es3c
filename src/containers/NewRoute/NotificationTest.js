const axios = require("axios");
  
const { invite } = require("./invite");
const { fetch_inbox } = require("./discover-inbox");

// Let's assume that we know the IRI of the invitee's calendar
const calendar_iri = "https://uo246355.solid.community/inbox";

// const store = $rdf.graph();
//const fetcher = new $rdf.Fetcher(store, timeout)

// First, fetch the inbox...
fetch_inbox(calendar_iri)
  .then(inbox => {
    // ...and then perform a POST to the target inbox to send
    // the invite defined in invite.js
    axios({
      method: "post",
      url: inbox,
      data: invite,
      headers: {
        "Content-type": "text/turtle",
        Slug: "We gonna parteyy"
      }
    })
      .then(response => {
        console.log(response.headers);
        if (response.status === 201) {
          document.getElementById("app").innerHTML = `
        <h1>Notification sent</h1>
        <div>
          Check the inbox of ${calendar_iri}: ${
            response.headers.location
          } has been created
        </div>
        `;
        }
      })
      .catch(error => {
        console.log(error);
      });
  })
  .catch(error => {
    console.log(error);
  });