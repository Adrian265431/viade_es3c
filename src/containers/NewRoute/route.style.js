import styled from "styled-components";

export const RouteWrapper = styled.section`
	display: flex;
	flex: 1 0 auto;
	flex-direction: row;
	background-color: #ffffff;
`;

export const Header = styled.div`
	max-width: 17%;
	align-items: top;
	justify-content: center;
	background: rgb(0, 77, 134);
	background: linear-gradient(180deg, rgba(0, 77, 134, 1) 0%, rgba(227, 222, 222, 1) 92%);
	padding: 30px 20px;
	p {
		color: white;
	}
`;

export const Input = styled.input`margin: 5px;`;

export const Button = styled.button`
	max-width: 128px;
	display: inline-block;

	&:first-child {
		margin-right: 10px;
	}
`;

export const TextArea = styled.textarea`resize: none;`;

export const DivForms = styled.div`margin: 1em;`;

export const InputSubmit = styled.input`float: right;`;

export const LabelInput = styled.label`color: white;`;

export const TitleRoute = styled.h3`
	text-align: center;
	color: white;
`;

export const UploaderFiles = styled.h3`
	margin-top: 1em;
	text-align: center;
`;

export const DivMin = styled.div``;

export const InputFile = styled.input``;

export const DivDivisor = styled.div`padding: 30px 20px;`;

export const RouteForm = styled.form`margin-bottom: 1em;`;
