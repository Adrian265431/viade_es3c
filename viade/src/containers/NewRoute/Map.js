import {Map, GoogleApiWrapper, Marker, Polyline} from 'google-maps-react';
import React from 'react';
import update from 'react-addons-update';

const mapStyles = {
    height: '300px',
};

export class MapContainer extends React.Component {

    constructor(props) {
        super(props);
    }

    sendData = () => {
        this.props.parentCallBack(this.state.markers);
    };

    state = {markers:[]};

    clickPoint = (event, map, clickEvent) => {
        let {markers} = this.state;
        markers = update(markers, {
            $push: [
                {
                    position: {
                        lat: clickEvent.latLng.lat(),
                        lng: clickEvent.latLng.lng()
                    },
                    defaultAnimation: 2,
                    key: Date.now(),
                },
            ],
        });
        this.setState({markers});
        this.sendData();
    };

    getLocation() {
        if (navigator.geolocation) {
            navigator.geolocation.getCurrentPosition((position) => {
                this.setState({
                    center: {
                        lat: position.coords.latitude,
                        lng: position.coords.longitude,
                    }
                })
            });
        }
    }

    draw() {
        let markers = [];
        for (let i = 0; i < this.state.markers.length; i++) {
            markers.push({lat: this.state.markers[i].position.lat, lng: this.state.markers[i].position.lng})
        }
        return markers;
    };

    render() {
        this.getLocation();
        return (
            <Map
                google={this.props.google}
                zoom={8}
                style={mapStyles}
                onClick={this.clickPoint}
                center={this.state.center}
            >

            {this.state.markers.map((marker) => {
                return (
                    <Marker position={{lat: marker.position.lat, lng: marker.position.lng, alt: marker.position.alt}}/>
                );
            })}

            <Polyline
                path={this.draw()}
                strokeColor="#01C9EA"
                strokeOpacity={0.8}
                strokeWeight={2}/>
            </Map>
        );
    }
}

export default GoogleApiWrapper({
    apiKey: 'AIzaSyB3kcMOsEbg2unbt5yGvqw4HxNlSLE-U00'
})(MapContainer);

