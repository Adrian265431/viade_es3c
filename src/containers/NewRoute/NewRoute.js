import React from "react";
import Map from "./Map";
import { errorToaster, successToaster } from "@utils";
import {
	Header,
	RouteWrapper,
	TextArea,
	DivForms,
	InputSubmit,
	LabelInput,
	TitleRoute,
	RouteForm
} from "./route.style";
import { viadeManager } from "@utils";
import { Route, Point, Multimedia } from "domain";
import { MultimediaComponent } from "../UploadMultimedia/multimedia.container";
import i18n from "i18n";

type Props = { webId: String };

class NewRoute extends React.Component {
	constructor({ webId }: Props) {
		super();
		this.webID = webId;
		this.handleSave = this.handleSave.bind(this);
		this.handleChange = this.handleChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
		this.title = React.createRef();
		this.descripton = React.createRef();
		this.state = {
			markers: null
		};
	}
	
	state = { markers: {}, image: {} };

	callBackFunction = (childData) => {
		this.setState({ markers: childData });
	};

	componentWillUnmount() {}

	handleChange(event) {
		this.setState({ value: event.target.value });
	}

	handleSubmit(event) {
		event.preventDefault();
		this.handleSave(event);
	}

	async handleSave(event) {
		if (this.title.current.value.length === 0) {
			errorToaster(i18n.t("newRoute.errorTitle"), "ERROR");
		} else if (this.descripton.current.value.length === 0) {
			errorToaster(i18n.t("newRoute.errorDescription"), "ERROR");
		} else if (this.state.markers === null || this.state.markers.length < 0) {
			errorToaster(i18n.t("newRoute.errorPoints"), "ERROR");
		} else {
			const points = [];
			for (let i = 0; i < this.state.markers.length; i++) {
				points.push(
					new Point(
						this.state.markers[parseInt(i)].position.lat,
						this.state.markers[parseInt(i)].position.lng,
						this.state.markers[parseInt(i)].position.alt
					)
				);
			}

			let author = this.webID.replace("https://", "");
			author = author.replace(".solid.community/profile/card#me", "");
			author = author.replace(".inrupt.net/profile/card#me", "");

			const multimedia = [];
			let filesFolder = document.getElementsByClassName("file-uploader--input");
			let filesMult = filesFolder[0].files;
			let url = this.webID.replace("profile/card#me", "public/viade/rawMedia/");
			for (let j = 0; j < filesMult.length; j++) {
				let name = filesMult[parseInt(j)].name.split(".")[0];
				var d = Date(Date.now());
				multimedia.push(new Multimedia(url + filesMult[parseInt(j)].name, d.toString(), author, name, null));
			}
			let route = new Route(this.title.current.value, author, this.descripton.current.value, points, multimedia);
			await viadeManager.addRoute(route, this.webID);
			successToaster(i18n.t("newRoute.successRoute"), i18n.t("newRoute.success"));
			setTimeout(function() {
				window.location.href = "#/myRoutes";
			}, 1000);
		}
		event.persist();
	}

	render(): React.ReactNode {
		return (
			<RouteWrapper data-testid="route-component">
				<Header>
					<TitleRoute>{i18n.t("newRoute.title")}</TitleRoute>
					<RouteForm id="routef">
						<DivForms>
							<LabelInput>
								{i18n.t("newRoute.name")}{" "}
								<input
									type="text"
									id="route_name"
									name="route_name"
									ref={this.title}
								/>
							</LabelInput>
						</DivForms>
						<DivForms>
							<LabelInput>
								{" "}
								{i18n.t("newRoute.description")}{" "}
								<TextArea
									type="text"
									id="description"
									name="description"
									rows="10"
									ref={this.descripton}
								/>{" "}
							</LabelInput>
						</DivForms>
					</RouteForm>
					<DivForms>
						<MultimediaComponent id={"input-img"} webId={`[${this.webId}]`} image="" />
					</DivForms>
					<DivForms>
						<InputSubmit
							type="submit"
							id="save_route"
							value={i18n.t("newRoute.btnSave")}
							form="routef"
							onClick={this.handleSubmit}
						/>
					</DivForms>
				</Header>
				<Map parentCallBack={this.callBackFunction} zoom={13} />
			</RouteWrapper>
		);
	}
}

export default NewRoute;
