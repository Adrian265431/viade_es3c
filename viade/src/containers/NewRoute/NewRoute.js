import React from 'react';
import {successToaster, errorToaster} from '@utils';
import {Loader} from '@util-components';
import Map from './Map';
import {
    Header,
    RouteWrapper,
    Button
} from './route.style';
import {Input} from "../TextEditor/text-editor.style";

function saveRoute() {

}

class NewRoute extends React.Component{

    state = {markers: {}};

    callBackFunction = (childData) => {
        this.setState({markers: childData})
        console.log(this.state.markers);
    };

    render(): React.ReactNode {
        return (
            <RouteWrapper data-testid="route-component">
                <Header>
                    <h5 className="text--white">Nombre:</h5>
                    <Input type="text" size="10" />
                    <Button id="saveRoute" onClick={saveRoute}>Guardar</Button>
                </Header>
                <Map parentCallBack = {this.callBackFunction} zoom={20}/>
            </RouteWrapper>
        );
    }


};

export default NewRoute;
