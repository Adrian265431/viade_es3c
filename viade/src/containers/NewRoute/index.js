import NewRoute from './route.container';

export default NewRoute;
