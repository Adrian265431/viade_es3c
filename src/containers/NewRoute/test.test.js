import React from "react";
import { shallow } from "enzyme";
import NewRoute from "./NewRoute";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("NewRoute", () => {
  let wrapper;
  let mockSubmit;
  beforeEach(() => {
    mockSubmit = jest.fn();
    wrapper = shallow(<NewRoute onClick={mockSubmit}/>);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("handleChange", () => {
    it("should call setState on title", () => {
      const mockEvent = {
        target: {
          markers: null,
          value: [
            { position: { lat: 43.354831, lng: -5.851303 } },
            { position: { lat: 43.35644, lng: -5.854693 } },
            { position: { lat: 43.361836, lng: -5.850547 } }
          ]
        }
      };
      const expected = {
        markers: null,
        value: [
          { position: { lat: 43.354831, lng: -5.851303 } },
          { position: { lat: 43.35644, lng: -5.854693 } },
          { position: { lat: 43.361836, lng: -5.850547 } }
        ]
      };
      wrapper.instance().handleChange(mockEvent);

      expect(wrapper.state()).toEqual(expected);
    });
  });


});

