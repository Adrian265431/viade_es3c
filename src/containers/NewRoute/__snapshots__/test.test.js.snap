// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewRoute should match the snapshot 1`] = `
<styled.section
  data-testid="route-component"
>
  <styled.div
    data-testid="route-header"
  >
    <styled.h3>
      newRoute.title
    </styled.h3>
    <styled.form
      id="routef"
    >
      <styled.div>
        <styled.label>
          newRoute.name
           
          <input
            id="route_name"
            name="route_name"
            type="text"
          />
        </styled.label>
      </styled.div>
      <styled.div>
        <styled.label>
           
          newRoute.description
           
          <styled.textarea
            id="description"
            name="description"
            rows="10"
            type="text"
          />
           
        </styled.label>
      </styled.div>
    </styled.form>
    <styled.div>
      <MultimediaComponent
        id="input-img"
        image=""
        webId="[undefined]"
      />
    </styled.div>
    <styled.div>
      <styled.input
        form="routef"
        id="save_route"
        onClick={[Function]}
        type="submit"
        value="newRoute.btnSave"
      />
    </styled.div>
  </styled.div>
  <Wrapper
    parentCallBack={[Function]}
    zoom={13}
  />
</styled.section>
`;
